src/vs/workbench/contrib/terminalContrib/suggest/test/browser/terminalCompletionService.test.ts

```diff
@@ -67,129 +95,247 @@ suite('TerminalCompletionService', () => {
-				provider,
-				kind: TerminalCompletionItemKind.Folder,
-				isDirectory: true,
-				isFile: false,
-				replacementIndex: 3,
-				replacementLength: 1 // replacing .
-			}]);
-		});
-		test('cd ./|', async () => {
+			const result = await terminalCompletionService.resolveResources(resourceRequestConfig, './', 2, provider);
+
+			assertCompletions(result, [
+				{ label: './', detail: 'test' },
+				{ label: './.hiddenFile', kind: TerminalCompletionItemKind.File },
+				{ label: './.hiddenFolder/', kind: TerminalCompletionItemKind.Folder },
+				{ label: './folder1/', kind: TerminalCompletionItemKind.Folder },
+				{ label: './file1.txt', kind: TerminalCompletionItemKind.File },
+				{ label: './../', kind: TerminalCompletionItemKind.Folder },
+			], { replacementIndex: 0, replacementLength: 2 });
+		});
+
+		test('./h| should handle hidden files and folders', async () => {
 			const resourceRequestConfig: TerminalResourceRequestConfig = {
 				cwd: URI.parse('file:///test'),
 				foldersRequested: true,
+				filesRequested: true,
 				pathSeparator
 			};
-			const result = await terminalCompletionService.resolveResources(resourceRequestConfig, 'cd ./', 5, provider);
-			assert.deepEqual(result, [{
-				label: `.${pathSeparator}folder1${pathSeparator}`,
-				provider,
-				kind: TerminalCompletionItemKind.Folder,
-				isDirectory: true,
-				isFile: false,
-				replacementIndex: 3,
-				replacementLength: 2 // replacing ./
-			}]);
-		});
-		test('cd ./f|', async () => {
+			const result = await terminalCompletionService.resolveResources(resourceRequestConfig, './h', 3, provider);
+
+			assertCompletions(result, [
+				{ label: './', detail: 'test' },
+				{ label: './.hiddenFile', kind: TerminalCompletionItemKind.File },
+				{ label: './.hiddenFolder/', kind: TerminalCompletionItemKind.Folder },
+				{ label: './folder1/', kind: TerminalCompletionItemKind.Folder },
+				{ label: './file1.txt', kind: TerminalCompletionItemKind.File },
+				{ label: './../', kind: TerminalCompletionItemKind.Folder },
+			], { replacementIndex: 0, replacementLength: 3 });
+		});
+	});
+	suite('resolveResources edge cases and advanced scenarios', () => {
+		setup(() => {
+			validResources = [];
+			childResources = [];
+		});
+
+		test('/usr/| Missing . should show correct results', async () => {
+			const resourceRequestConfig: TerminalResourceRequestConfig = {
+				cwd: URI.parse('file:///'),
+				foldersRequested: true,
+				pathSeparator
+			};
+			validResources = [URI.parse('file:///')];
+			childResources = [
+				{ resource: URI.parse('file:///usr/'), isDirectory: true },
+				{ resource: URI.parse('file:///home/'), isDirectory: true }
+			];
+			const result = await terminalCompletionService.resolveResources(resourceRequestConfig, '/usr/', 5, provider);
+
+			assertCompletions(result, [
+				{ label: '/usr/' },
+				{ label: './usr/' },
+				{ label: './home/' },
+				{ label: '/usr/../' }
+			], { replacementIndex: 0, replacementLength: 5 });
```

Comment:
This should actually not include the `.` ones here since `/` is how you signal an absolute path when not on Windows.